core/Updater.php

```diff
@@ -279,6 +279,15 @@ public function update($componentName)
 
         $this->executeListenerHook('onComponentUpdateFinished', array($componentName, $updatedVersion, $warningMessages));
 
+        /**
+         * Event triggered after a component has been updated.
+         *
+         * @param string $componentName 'core', or plugin name
+         * @param string $updatedVersion version updated to
+         * @param array  $warningMessages warnings occurred during update
+         */
+        Piwik::postEvent('Updater.componentUpdated', array($componentName, $updatedVersion, $warningMessages));
```

Comment:
the current implementation doesn't use the `onComponentUpdateFinished` hook at all. As far as I have seen we only have the CLI observer that uses some hooks. But that one is unused. Simply removing it would maybe make the observer kind of incomplete for possible new use cases. Completely removing the observer and replacing it using events, might not be the best solution, as well.